src/arrayMethodReduce.js

```diff
@@ -6,6 +6,20 @@
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
     // write code here
+    let result = initialValue;
+    let index = 0;
+
+    if (initialValue === undefined) {
+      result = this[0];
+      index = 1;
+    }
+
+    while (index < this.length) {
+      result = callback(result, this[index], index, this);
+      index++;
+    }
```

Comment:
`for(...)` loop would be appropriate here.
Thumb rule is: 
1. If you know the exact number of iterations (like in this case) - use `for` with index
2. If you don't know the exact number of iterations, or it depends on some other conditions - use `while` or `do...while`.